### DISCUSSION TITLE
Question about `referrer_path` in mount mode

### DISCUSSION THREAD

**Discussion by daya0576** [zauberzeug/nicegui#2711] [Q&A] [answered]
### Question about `referrer_path` in mount mode

Hi,

Let's start nicegui app with `mount_path` param:
```python
ui.run_with(app, mount_path="gui")
```

Redirect to previous page after login:
```python
# step1: remember current page
app.storage.user["referrer_path"] = request.url.path

# step2: redirect to previous page
ui.navigate.to(app.storage.user.get("referrer_path", "/"))
```

 ❌ After login, it goes to `/gui/gui/` (instead of `/gui`)

🤔 How can I resolve this problem? Thanks!

**Comment by rodja** [comment, marked as answer]
`request.url.path` contains the full path, in your case `/gui`. But as your app is mounted on `/gui` itself, the `ui.navigate.to` will prepend this to every navigation request starting with `/`. This is correct, because the sub-app should not need to know about the mount-path.
Your problem is the "remembering of the current page". It should substract `request.scope['root_path']` before storing it.

**Reply by daya0576** [reply]
Thanks! It's working now.
